### PR TITLE
Added test code + fetch route

### PR DESCRIPTION
**Pull request by mathisve**
Added some questionable (but working) test code.
You can run it by doing `go test`.

Also added the `/V1/fetch` route to get the latest 10 bumped servers.

### CODE REVIEW

File: api/src/bump_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+	go HandleRequests()
+}
+
+func sendInt(guildId int) (br BumpResponse) {
+	reqBody, _ := json.Marshal(map[string]int{
+		"guildId": guildId,
+	})
+
+	resp, _ := http.Post("http://localhost:8080/V1/bump", "application/json", bytes.NewBuffer(reqBody))
```
Comment (Travis-Owens) [inline api/src/bump_test.go]: The port "8080" should reflect the constant set in "[/api/src/main.go](https://github.com/Coding-Web-Community/CodingBump/blob/1e1261cbd1450db1d13e7ec5ee425a7c49dbd44e/api/src/main.go#L14)". If this becomes a distributed application, then "localhost" will become problematic, thus this should also be a constant.

https://github.com/Coding-Web-Community/CodingBump/blob/1e1261cbd1450db1d13e7ec5ee425a7c49dbd44e/api/src/main.go#L14

File: api/src/bump_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+	return br
+}
+
+func sendString(guildId string) (br BumpResponse) {
+	reqBody, _ := json.Marshal(map[string]string{
+		"guildId": guildId,
+	})
+
+	resp, _ := http.Post("http://localhost:8080/V1/bump", "application/json", bytes.NewBuffer(reqBody))
```
Comment (Travis-Owens) [inline api/src/bump_test.go]: See previous comment.

File: api/src/bump_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	if resp.Message != expected.Message {
+		t.Errorf("Messages were not equal: %v != %v", resp.Message, expected.Message)
+	}
+}
+
+func TestFetch(t *testing.T) {
+	var fr FetchResponse
+	guilds, _ := LoadStore()
+
+	resp, _ := http.Get("http://localhost:8080/V1/fetch")
```
Comment (Travis-Owens) [inline api/src/bump_test.go]: See previous comment.

File: api/src/main.go [hunk 4]
```diff
@@ -13,10 +13,15 @@ import (
 const (
 	PORT            = ":8080"
 	BUMP_INTERVAL   = 60 // 1 minute in seconds
+	TEST_INTERVAL   = 2
```
Comment (Travis-Owens) [inline api/src/main.go]: This constant has not been implemented/used.

File: api/src/main.go [hunk 5]
```diff
@@ -13,10 +13,15 @@ import (
...
 	STORE_FILE_NAME = "store.json"
 )
 
-var gs GuildStore
+var (
+	TempTestInterval = 0    // used to set lower interval during testing
```
Comment (Travis-Owens) [inline api/src/main.go:21]: Should this reflect the constant "TEST_INTERVAL"?

**Review by Travis-Owens — COMMENTED** [review]
@Mathisco-01, overall good work.  Only a couple of minor things that need to be reviewed.

Both the "TEST_INTERVAL" and "PORT" constants are not being used throughout the source code.

Outside of this PR, we should consider replacing the occurrences of "localhost" with a constant placeholder to facilitate distributing this application in a cloud environment.

File: api/src/bump_test.go (hunk 1 above)
Comment (mathisve) [reply api/src/bump_test.go]: Good call!
Changed the URL to a global variable too!

File: api/src/bump_test.go (hunk 3 above)
Comment (mathisve) [reply api/src/bump_test.go]: Fixed

File: api/src/main.go (hunk 5 above)
Comment (mathisve) [reply api/src/main.go:21]: the global variable `TEST_INTERVAL` has been removed.
This is now the only variable used when changing the interval for testing

**Comment by mathisve** [comment]
I almost forgot to mention but I added documentation for the fetch route.

**Review by Travis-Owens — COMMENTED** [review]
@Mathisco-01, good job on resolving the issues. Looking forward, we should add more commenting to the source code, as this will increase readability and maintainability.

**Review by Travis-Owens — APPROVED** [review]

**Review by blacksmithop — APPROVED** [review]
Looks good 👍

**Pull request merged**